cmake/SetupCompilers.cmake

```diff
@@ -103,37 +103,46 @@ if ( MSVC )
 endif()
 
 if (RAJA_ENABLE_CUDA)
-  set(RAJA_NVCC_STD "c++11")
-  # When we require cmake 3.8+, replace this with setting CUDA_STANDARD
-  if(CUDA_VERSION_MAJOR GREATER "8")
+
+  if ( NOT DEFINED RAJA_NVCC_STD ) 
+    set(RAJA_NVCC_STD "c++11")
+    # When we require cmake 3.8+, replace this with setting CUDA_STANDARD
+    if(CUDA_VERSION_MAJOR GREATER "8")
       execute_process(COMMAND ${CUDA_TOOLKIT_ROOT_DIR}/bin/nvcc -std c++14 -ccbin ${CMAKE_CXX_COMPILER} . 
-                    ERROR_VARIABLE TEST_NVCC_ERR
-                    OUTPUT_QUIET)
-    if (NOT TEST_NVCC_ERR MATCHES "flag is not supported with the configured host compiler")
-      set(RAJA_NVCC_STD "c++14")
+                      ERROR_VARIABLE TEST_NVCC_ERR
+                      OUTPUT_QUIET)
+      if (NOT TEST_NVCC_ERR MATCHES "flag is not supported with the configured host compiler")
+        set(RAJA_NVCC_STD "c++14")
+      endif()
+    else()
     endif()
-  else()
   endif()
-  if(CMAKE_BUILD_TYPE MATCHES Release)
-    set(RAJA_NVCC_FLAGS -O2; -restrict; -arch ${RAJA_CUDA_ARCH}; -std ${RAJA_NVCC_STD}; --expt-extended-lambda; -ccbin; ${CMAKE_CXX_COMPILER} CACHE LIST "")
-  elseif(CMAKE_BUILD_TYPE MATCHES Debug)
-    set(RAJA_NVCC_FLAGS -g; -G; -O0; -restrict; -arch ${RAJA_CUDA_ARCH}; -std  ${RAJA_NVCC_STD}; --expt-extended-lambda; -ccbin ${CMAKE_CXX_COMPILER} CACHE LIST "")
-  elseif(CMAKE_BUILD_TYPE MATCHES MinSizeRel)
-    set(RAJA_NVCC_FLAGS -Os; -restrict; -arch ${RAJA_CUDA_ARCH}; -std ${RAJA_NVCC_STD}; --expt-extended-lambda; -ccbin; ${CMAKE_CXX_COMPILER} CACHE LIST "")
-  else(CMAKE_BUILD_TYPE MATCHES RelWithDebInfo)
-    set(RAJA_NVCC_FLAGS -g; -G; -O2; -restrict; -arch ${RAJA_CUDA_ARCH}; -std  ${RAJA_NVCC_STD}; --expt-extended-lambda; -ccbin ${CMAKE_CXX_COMPILER} CACHE LIST "")
-  endif()
-  if(RAJA_ENABLE_COVERAGE)
-    if (CMAKE_CXX_COMPILER_ID MATCHES GNU)
-      message(INFO "Coverage analysis enabled")
-      set(RAJA_NVCC_FLAGS ${RAJA_NVCC_FLAGS}; -Xcompiler -coverage; -Xlinker -coverage)
-      set(CMAKE_EXE_LINKER_FLAGS "-coverage ${CMAKE_EXE_LINKER_FLAGS}")
-    else()
-      message(WARNING "Code coverage specified but not enabled -- GCC was not detected")
+
+  if (NOT RAJA_HOST_CONFIG_LOADED)
```

Comment:
@davidbeckingsale why do we want to introduce this CMake variable instead of using whether the RAJA_NVCC_FLAGS variable directly as in my original PR?  This seems to make life harder for users, to me at least.